### ISSUE TITLE
Re11-worldwise > src > components > CityItem | Navigate component not working as expected

### ISSUE THREAD

**Issue by devYuraKim** [devYuraKim/react#17]
I wanted to set the code up so that when the user clicks on the `<li>` element, the webpage redirects to `currentURL/city.id` and then renders the `<City>` component.

`CityItem` component
```
function CityItem({ city }) {
  function handleClick() {
    return <Navigate to={city.id} element={<City />} />;
  }
  return (
    <li className={styles.cityItem} onClick={handleClick}>
//...remaining code
```

---

Based on the fact this code snippet worked, I expected the above code to work as well.

`App` component
```
<Route index element={<Navigate to="cities" />} />
     <Route path="cities" element={<CityList cities={cities}  isLoading={isLoading} />} />
```

**Comment by devYuraKim** [comment]
TL;DR: know when to use `<Navigate>` or `useNavigate()`
---
- **Issue1: `<Navigate>` is used in JSX**
`<Navigate>` is a React component meant to be used in JSX to trigger navigation when the component is rendered. It’s not intended to be used inside event handlers like handleClick. React components can't be used directly in this way to handle user interactions.

- **Issue2: `handleClick` needs functions**
Event handlers like onClick expect functions that execute code but don't render JSX.

- Solution
```
function CityItem({ city }) {
  const navigate = useNavigate();

  function handleClick() {
    navigate(`/path-to-city/${city.id}`);
  }

  return (
    <li className={styles.cityItem} onClick={handleClick}>
//...remaining code
```

**Comment by devYuraKim** [comment]
TL;DR: `<Navigate>` requires **a predefined path** in the routing configuration that is associated with a component(element)
---

```
<Route index element={<Navigate to="cities" />} />
```
worked because: 
1. A path was already defined with `/cities`.
2. The `/cities` path was linked to a component to be rendered (`<CityList />`).
3. By rendering the `<Navigate>` component, the application redirected to the `/cities` path, which in turn rendered the component associated with that path.

---

Even for **DYNAMIC NAVIGATION**, the basic path must be defined in the routes:

```
<Navigate to={`/${cityId}`} />
```

```
<Routes>
  <Route path="/:cityId" element={<City />} />
</Routes>
```
